flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/SnapshotManager.java

```diff
@@ -214,13 +217,16 @@ public Optional<Snapshot> latestSnapshotOfUser(String user) {
     public Long readHint(String fileName) {
         Path snapshotDir = snapshotDirectory();
         Path path = new Path(snapshotDir, fileName);
-        try {
-            if (path.getFileSystem().exists(path)) {
+        int retryNumber = 0;
+        while (retryNumber++ < READ_HINT_RETRY_NUM) {
+            try {
                 return Long.parseLong(FileUtils.readFileUtf8(path));
+            } catch (IOException ignored) {
```

Comment:
`Exception` is more safe, after all, read hint is just an optimization

Reply:
Done.